primitives/calc.yaml

```diff
@@ -869,4 +869,89 @@ definitions:
      lscpu_flags: [ ]
      implementation: return vec_a / vec_b;
 #INTEL - FPGA
+---
+primitive_name: "hor"
+brief_description: "Operate horizontal or on vector register"
+parameters:
+   - ctype: "const typename Vec::register_type"
+     name: "vec"
+     description: "Operating Vector"
+returns:
+   ctype: "typename Vec::base_type"
+   description: "Value containing result of the horizontal or"
+testing:
+   - test_name: "with_only_zero"
+     requires: ["set1", "loadu", "storeu"]
+     includes: ["<cstddef>"]
+     implementation: |
+        using T = typename Vec::base_type;
+        testing::test_memory_helper_t<Vec> test_helper{1, false};
+        auto reference_result_ptr = test_helper.result_ref();
+        auto test_result_ptr = test_helper.result_target();
+        auto vec = set1<Vec>(0);
+        *reference_result_ptr = 0;
+        *test_result_ptr = hor<Vec>(vec);
+        test_helper.synchronize();
+        return test_helper.validate();
+   - test_name: "with_only_one"
+     requires: ["set1", "loadu", "storeu"]
+     includes: ["<cstddef>"]
+     implementation: |
+        using T = typename Vec::base_type;
+        testing::test_memory_helper_t<Vec> test_helper{1, false};
+        auto reference_result_ptr = test_helper.result_ref();
+        auto test_result_ptr = test_helper.result_target();
+        auto vec = set1<Vec>(1);
+        *reference_result_ptr = 1;
+        *test_result_ptr = hor<Vec>(vec);
+        test_helper.synchronize();
+        return test_helper.validate();
+   - test_name: "with_sequence"
+     requires: ["sequence", "loadu", "storeu"]
+     includes: ["<cstddef>"]
+     implementation: |
+        using T = typename Vec::base_type;
+        testing::test_memory_helper_t<Vec> test_helper{1, false};
+        auto reference_result_ptr = test_helper.result_ref();
+        auto test_result_ptr = test_helper.result_target();
+        auto vec = sequence<Vec>();
+        *reference_result_ptr = (bool) (Vec::vector_element_count() - 1);
+        *test_result_ptr = hor<Vec>(vec);
+        test_helper.synchronize();
+        return test_helper.validate();
+definitions:
+#INTEL - AVX512
+   - target_extension: "avx512"
+     ctype: ["uint8_t", "int8_t", "uint16_t", "int16_t", "int32_t", "uint32_t", "int64_t", "uint64_t"]
+     lscpu_flags: ['avx512f', 'avx512bw']
+     implementation: return (_mm512_cmpneq_epi{{intrin_tp[ctype][1]}}_mask(vec, _mm512_setzero_si512()) != 0);
```

Comment:
1. I don't understand the provided implementation. Let's assume we have uint64_t values in our 512-bit SIMD reg:
```
using SIMD = simd<uint64t_t, avx512>;
auto data = set<SIMD>(0x4, 0x40, 0x400, 0x4000, 0x8, 0x80, 0x800, 0x8000);
auto result = hor<SIMD>(data);

```
The result should be `0xCCCC`.
However, `(_mm512_cmpneq_epi64_mask(data, _mm512_setzero_si512())` would be **0xFF**, since all values are equal 0. Compared with **!=0**, the final result would be **true** (or **1** through the implicit cast)... 
This doesn't look right, or do I miss out on something?

2. Can you please split up the definitions into `[(u)int(8|16)]` and `[(u)int(32|64)]` since the latter one only needs _AVX512F_. Consequently, for 4- and 8-byte values we could provide a definition if the CPU does not support _AVX512BW_.